File: frontend/health-enforcement-ui/src/api/system/industry.js
import request from '@/utils/request'
import { parseStrEmpty } from "@/utils/ruoyi";

// 查询行业分类列表
export function listIndustry(query) {
  return request({
    url: '/system/industry/list',
    method: 'get',
    params: query
  })
}

// 查询行业分类详细
export function getIndustry(industryId) {
  return request({
    url: '/system/industry/' + parseStrEmpty(industryId),
    method: 'get'
  })
}

// 新增行业分类
export function addIndustry(data) {
  return request({
    url: '/system/industry',
    method: 'post',
    data: data
  })
}

// 修改行业分类
export function updateIndustry(data) {
  return request({
    url: '/system/industry',
    method: 'put',
    data: data
  })
}

// 删除行业分类
export function deleteIndustry(industryId) {
  return request({
    url: '/system/industry/' + industryId,
    method: 'delete'
  })
}

File: app/admin/router/system/industry.go
package system

import (
	"github.com/gin-gonic/gin"
	"haocean/health-enforcement/app/admin/api/system"
	"haocean/health-enforcement/app/core/utils/jwt"
)

// InitIndustry 初始化行业分类路由
func InitIndustry(e *gin.Engine) {
	v := e.Group("system")
	{
		v.GET("/industry/list", system.ListIndustry)
		v.GET("/industry/:id", system.GetIndustry)

		auth := v.Group("")
		auth.Use(jwt.JWTAuthMiddleware())
		{
			auth.POST("/industry", system.AddIndustry)
			auth.PUT("/industry", system.UpdateIndustry)
			auth.DELETE("/industry/:ids", system.DeleteIndustry)
		}
	}
}
